pkg/contracts/src/RegistryCommunityV0_0.sol

```diff
@@ -0,0 +1,613 @@
+import {UUPSUpgradeable} from "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+
+import {OwnableUpgradeable} from "openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import {ReentrancyGuardUpgradeable} from
+    "openzeppelin-contracts-upgradeable/contracts/security/ReentrancyGuardUpgradeable.sol";
+import {AccessControlUpgradeable} from
+    "openzeppelin-contracts-upgradeable/contracts/access/AccessControlUpgradeable.sol";
+
+import {IAllo} from "allo-v2-contracts/core/interfaces/IAllo.sol";
+import {Clone} from "allo-v2-contracts/core/libraries/Clone.sol";
+import {IRegistry, Metadata} from "allo-v2-contracts/core/interfaces/IRegistry.sol";
+import {FAllo} from "./interfaces/FAllo.sol";
+import {ISafe} from "./interfaces/ISafe.sol";
+import {RegistryFactory} from "./RegistryFactory.sol";
+import {IPointStrategy, CVStrategy, StrategyStruct} from "./CVStrategy.sol";
+
+contract RegistryCommunityV0_0 is
+    OwnableUpgradeable,
+    UUPSUpgradeable,
+    ReentrancyGuardUpgradeable,
+    AccessControlUpgradeable
+{
+    using ERC165Checker for address;
+    using SafeERC20 for IERC20;
+    using Clone for address;
+
+    /// @notice The native address to represent native token eg: ETH in mainnet
+    address public constant NATIVE = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    /// @notice The precision scale used in the contract to avoid loss of precision
+    uint256 public constant PRECISION_SCALE = 10 ** 4;
+    /// @notice The maximum fee that can be charged to the community
+    uint256 public constant MAX_FEE = 10 * PRECISION_SCALE;
+
+    /// @notice Enable or disable the kick feature
+    bool public isKickEnabled;
+
+    /// @notice The amount of tokens required to register a member
+    uint256 public registerStakeAmount;
+    /// @notice The fee charged to the community for each registration
+    uint256 public communityFee;
+    /// @notice The profileId of the community in the Allo Registry
+    bytes32 public profileId;
+
+    /// @notice The address that receives the community fee
+    address public feeReceiver;
+    /// @notice The address of the registry factory
+    address public registryFactory;
+    /// @notice The address of the strategy template
+    address public strategyTemplate;
+    /// @notice The nonce used to create new strategy clones
+    uint256 public cloneNonce;
+    /// @notice The address of the pending council safe owner
+    address payable public pendingCouncilSafe;
+
+    /// @notice The Registry Allo contract
+    IRegistry public registry;
+    /// @notice The token used to stake in the community
+    IERC20 public gardenToken;
+    /// @notice The council safe contract address
+    ISafe public councilSafe;
+    /// @notice The Allo contract address
+    FAllo public allo;
+
+    /// @notice The community name
+    string public communityName;
+    /// @notice The covenant IPFS hash of community
+    string public covenantIpfsHash;
+
```

Comment:
Do you think we need gaps with that implementation ?

Reply:
Gaps always in the end of file